src/pages/ContentScripts/features/network-adj-show/index.tsx

```diff
@@ -0,0 +1,136 @@
+
+import { render, Container } from 'react-dom';
+import $ from 'jquery';
+import isPerceptor from '../../../../helpers/is-perceptor';
+import { getRepoName, isPublicRepoWithMeta } from '../../../../helpers/get-repo-info';
+import { getRepoNetwork, getDeveloperNetwork } from '../../../../api/repo';
+import React, { useEffect, useRef } from 'react';
+
+const featureId = features.getFeatureID(import.meta.url);
+let repoName: string;
+let repoNetworks: any;
+let developerNetworks: any;
+let repoEdges: any;
+let developerEdges: any;
+const allNodes = new Set<string>();
+
+const getData = async () => {
+  repoNetworks = await getRepoNetwork(repoName);
+  developerNetworks = await getDeveloperNetwork(repoName);
+  repoEdges = repoNetworks.edges;
+  developerEdges = developerNetworks.edges;
+};
+
+function onHover(nodeName: string): [string, number][] {
+  let adjacentNodes: [string, number][] = [];
+  if (nodeName.includes('/')) {
+    repoEdges.forEach((edge: [string, string, number]) => {
+      if (edge[0] === nodeName && edge[2] >= 10) {
+        adjacentNodes.push([edge[1], edge[2]]);
+      } else if (edge[1] === nodeName && edge[2] >= 10) {
+        adjacentNodes.push([edge[0], edge[2]]);
+      }
+    });
+  } else {
+    developerEdges.forEach((edge: [string, string, number]) => {
+      if (edge[0] === nodeName && edge[2] >= 2.5) {
+        adjacentNodes.push([edge[1], edge[2]]);
+      } else if (edge[1] === nodeName && edge[2] >= 2.5) {
+        adjacentNodes.push([edge[0], edge[2]]);
+      }
+    });
+  }
+  // console.log(adjacentNodes);
+  return adjacentNodes;
+}
+
+function addTable(adjacentNodes, nodeListSpan) {
+  const table = document.createElement('table');
+  table.style.borderCollapse = 'collapse';
+  table.style.width = '100%';
+
+  // 添加表头
+  const headerRow = document.createElement('tr');
+  const headerKey = document.createElement('th');
+  headerKey.innerText = 'Related Repository';
+  headerKey.style.border = '1px solid white';
+  // headerKey.style.backgroundColor = '#f2f2f2';
+  const headerValue = document.createElement('th');
+  headerValue.innerText = 'Relation value';
+  headerValue.style.border = '1px solid white';
+  // headerValue.style.backgroundColor = '#f2f2f2';
+  headerRow.appendChild(headerKey);
+  headerRow.appendChild(headerValue);
+  table.appendChild(headerRow);
+
+  // 添加数据行
+  adjacentNodes.forEach((item: [string, number]) => {
+    const row = document.createElement('tr');
+    const cellKey = document.createElement('td');
+    cellKey.innerText = item[0];
+    cellKey.style.border = '1px solid white';
+    const cellValue = document.createElement('td');
+    cellValue.innerText = item[1].toString();
+    cellValue.style.border = '1px solid white';
+    row.appendChild(cellKey);
+    row.appendChild(cellValue);
+    table.appendChild(row);
+  });
+
+  // 将表格插入到指定元素中
+  for (const element of nodeListSpan) {
+    element.innerHTML = '';
+    element.appendChild(table);
+  }
+}
```

Comment:
Can we make this a separate component? It would be easy to maintain.

Reply:
Actually, I haven't learned much about front-end content, this is just an assignment for the course. I modified the code to present the table in the form of a view and it ran successfully, but there are still warnings when I check the console, and I don't know how to fix that.

thank u for ur help

![image](https://github.com/user-attachments/assets/a5e89bb4-d1f2-4dff-a4b6-b3c6bcd630b7)